pkg/simplewal/simplewal.go

```diff
@@ -150,7 +151,7 @@ func (w *WAL) loadAll(forEach func(index t.RetentionIndex, p *eventpb.Event)) er
 
 	for i := firstIndex; i <= lastIndex; i++ {
 		data, err := w.log.Read(i)
-		if err != nil {
+		if err != nil && !errors.Is(err, io.EOF) {
```

Comment:
I think EOF is still an error here. The log contains metadata about the `lastIndex` and that should be present. If it's not, it's an error.

Also, on line 149, there seems to be a copy-paste error. I believe it should sah "could not read last index".

Reply:
fixed